File: src/main/org/nlogo/api/World3D.java
package org.nlogo.api;

public interface World3D extends World
{
	int worldDepth() ;
	Protractor3D protractor3D() ;
	int minPzcor() ;
	int maxPzcor() ;
	double wrappedObserverZ( double z ) ;
	double wrapZ( double z ) ; // World3D always wraps at present, so no AgentException - ST 3/3/09
	double followOffsetZ() ;
	Patch getPatchAt( double x , double y , double z ) throws AgentException ;
}

File: headless/src/main/org/nlogo/api/World3D.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

trait World3D extends World {
  def worldDepth: Int
  def protractor3D: Protractor3D
  def minPzcor: Int
  def maxPzcor: Int
  def wrappedObserverZ(z: Double): Double
  def wrapZ(z: Double): Double  // World3D always wraps at present, so no AgentException - ST 3/3/09
  def followOffsetZ: Double
  @throws(classOf[AgentException])
  def getPatchAt(x: Double, y: Double, z: Double): Patch
}
